codebasin/source.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (C) 2019-2024 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+from pathlib import Path
+from typing import Union
+
+
+def is_source_file(filename: Union[str, os.PathLike]) -> bool:
+    """
+    Parameters
+    ----------
+    filename: Union[str, os.Pathlike]
+        The filename of a potential source file.
+
+    Returns
+    -------
+    bool
+        True if the file ends in a recognized extension and False otherwise.
+        Only files that can be parsed correctly have recognized extensions.
+
+    Raises
+    ------
+    TypeError
+        If filename is not a string or Path.
+    """
+    if not (isinstance(filename, str) or isinstance(filename, Path)):
+        raise TypeError("filename must be a string or Path")
+
+    extension = Path(filename).suffix
+    supported_extensions = [
+        ".f90",
+        ".F90",
```

Comment:
Do you want to just check with `str.lower` so it's case insensitive?

Reply:
I don't think we want to do that in general, because case might be important.  For Fortran, lower- and upper-case typically denote different forms of source code (yes, really).